sql/resource_sql_alerts.go

```diff
@@ -72,29 +73,38 @@ func (a *AlertEntity) toEditAlertApiObject(s map[string]*schema.Schema, data *sc
 func (a *AlertEntity) fromAPIObject(apiAlert *sql.Alert, s map[string]*schema.Schema, data *schema.ResourceData) error {
 	a.Name = apiAlert.Name
 	a.Parent = apiAlert.Parent
-	a.QueryId = apiAlert.Query.Id
+	if apiAlert.Query != nil {
+		a.QueryId = apiAlert.Query.Id
+	} else {
+		log.Printf("[WARN] Query object is nil in alert '%s' (id: %s) ", apiAlert.Name, apiAlert.Id)
```

Comment:
I think we should use [tflog.Warn](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-log/tflog#example-Warn) for logging in the provider. If we're not calling this from `terraform` directly, we may need to call `SetProviderRootLogger` from github.com/hashicorp/terraform-plugin-log/internal/logging.

Reply:
It works fine, taking into account `TF_LOG`, etc. The biggest pain point right now is not logging REST API payloads

Reply:
That issue should be fixed by https://github.com/databricks/terraform-provider-databricks/pull/2747.

Does this print logs even when the user hasn't configured them?

Reply:
yes, right now it's the default behavior. I'll file a separate GH Issue to rework logging in exporter.

Reply:
We have them everywhere in the provider. I.e., https://github.com/databricks/terraform-provider-databricks/blob/master/clusters/clusters_api.go#L683

Reply:
Oh I see... then let's leave it as is, and we can add a lint rule at the same time that we actually address this issue.